packages/astro/src/core/preview/static-preview-server.ts

```diff
@@ -72,8 +72,6 @@ export default async function createStaticPreviewServer(
 		host: getResolvedHostForHttpServer(settings.config.server.host),
 		port: settings.config.server.port,
 		closed,
-		// In Vite 5, `httpServer` may be a `Http2SecureServer`, but we know we are only starting a HTTP server
-		// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
 		server: previewServer.httpServer as http.Server,
```

Comment:
Is there a world where we'd want to support http for the preview server? Unrelated to this, just wondering.

Reply:
We could! At the moment we have control over the config passed to Vite's `preview()` so we know this will always be a http server. If we open up to the config from `astro.config.mjs` in the future, they can create https servers.